blob/blob_test.go

```diff
@@ -132,27 +129,72 @@ func TestListIterator(t *testing.T) {
 	}
 }
 
+// Verify that ListPage works even if driver.ListPaged returns empty pages.
+func TestListPage(t *testing.T) {
+	ctx := context.Background()
+	want := [][]string{{"a", "b"}, {"c", "d"}, {"e"}}
+	db := &fakeLister{
+		pages:         [][]string{{}, {"a", "b"}, {}, {}, {"c"}, {}, {"d"}, {}, {}, {"e"}},
+		wantPageSizes: []int{2, 2, 2, 2, 2, 1, 1, 2, 2, 2},
+	}
+	b := NewBucket(db)
+	defer b.Close()
+
+	nextToken := FirstPageToken
+	got := [][]string{}
+	for {
+		page, token, err := b.ListPage(ctx, nextToken, 2, nil)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(page) > 0 {
```

Comment:
But if `len(page) == 0` we continue looping?

This makes me realize it's not clear how to treat all return values of `ListPage` consistently. `err == io.EOF` signals "done", but also when the page is empty we're done, and also when token is nil we're done.

`ListPage` should document which is true when a bit more precisely so there would be a very obvious way of writing this loop.

Reply:
The conditional here was not necessary, removed.

The primary signal for ListPage being done is the empty token being returned. You can call it one more time if you want, but it will return an error (io.EOF). I tried to clarify the documentation a bit.